### PR TITLE
Add Document post-it perspective #LMR-244

### PR DESCRIPTION
**Pull request by justmesr**

### CODE REVIEW

File: src/main/scripts/src/app/app.module.ts [hunk 1]
```diff
@@ -34,6 +35,7 @@ import {DocumentsModule} from './documents/documents.module';
   imports: [
     BrowserModule,
     BrowserAnimationsModule,
+    HttpClientModule,
     CoreModule,
```
Comment (kubedo8) [inline src/main/scripts/src/app/app.module.ts]: This module is already in `CoreModule`

File: src/main/scripts/data/someicons.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  "fa-trophy",
+  "fa-bell",
+  "fa-flag",
+  "fa-android",
+  "fa-facebook-square",
+  "fa-twitter",
+  "fa-youtube-play",
+  "fa-music",
+  "fa-picture-o"
+]
```
Comment (kubedo8) [inline src/main/scripts/data/someicons.json]: After my commit will be merged (today), you should use constants in future `Const.ts` file

**Comment by kubedo8** [comment]
Add some screenshot or gif to demonstrate functionality

File: src/main/scripts/src/app/shared/perspectives/documents/post-it/post-it-documents-perspective.component.ts [hunk 3]
```diff
@@ -18,23 +18,150 @@
...
+              private workspaceService: WorkspaceService,
+              private route: ActivatedRoute,
+              private localService: LocalService) {
+  }
+
+  public ngOnInit() {
+    // initialize workspace if needed
+    let setWorkspaceParams = (params: ParamMap) => {
+      this.workspaceService.projectCode = params.get('projectCode');
+      this.workspaceService.organizationCode = params.get('organizationCode');
```
Comment (livthomas) [inline src/main/scripts/src/app/shared/perspectives/documents/post-it/post-it-documents-perspective.component.ts]: Why do you assign this to a variable if it is used only once?

File: src/main/scripts/src/app/shared/perspectives/documents/post-it/post-it-documents-perspective.component.ts [hunk 4]
```diff
@@ -18,23 +18,150 @@
...
+    };
+
+    if (!this.workspaceService.isWorkspaceSet()) {
+      this.route.paramMap.subscribe(setWorkspaceParams);
+    }
+
+    // get collection
```
Comment (livthomas) [inline src/main/scripts/src/app/shared/perspectives/documents/post-it/post-it-documents-perspective.component.ts]: You can extract some parts of `ngOnInit` method to multiple methods in order to increase code readability and avoid comments.

File: src/main/scripts/src/app/shared/perspectives/documents/post-it/post-it-documents-perspective.component.ts [hunk 5]
```diff
@@ -18,23 +18,150 @@
...
+        this.icons = icons;
+      });
+
+    this.localService.getColors()
+      .subscribe((colors: string[]) => {
+        this.colors = colors;
+      });
+  }
+
+  public getRandomElement = (array: any[]) => array[Math.floor(Math.random() * array.length)];
```
Comment (livthomas) [inline src/main/scripts/src/app/shared/perspectives/documents/post-it/post-it-documents-perspective.component.ts]: Would not it be more readable if this was defined as a normal function?

File: src/main/scripts/src/app/shared/perspectives/icon-picker/icon-picker.component.ts [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+  }
+
+  public selectColor(newColor: string): void {
+    this.selectedColor = newColor;
+    this.colorChange.emit(newColor);
+
+    this.change.emit(`selected ${newColor}`);
+  }
+
+  //=====================================================================
```
Comment (livthomas) [inline src/main/scripts/src/app/shared/perspectives/icon-picker/icon-picker.component.ts]: Why is this separator here? Should not rather this component be split into multiple child components?

File: src/main/scripts/src/app/shared/perspectives/documents/post-it/post-it-documents-perspective.component.ts (hunk 3 above)
Comment (justmesr) [inline src/main/scripts/src/app/shared/perspectives/documents/post-it/post-it-documents-perspective.component.ts]: I found it more readable that way, at the time.

File: src/main/scripts/src/app/shared/perspectives/icon-picker/icon-picker.component.ts (hunk 6 above)
Comment (justmesr) [inline src/main/scripts/src/app/shared/perspectives/icon-picker/icon-picker.component.ts]: I could split it into color and icon picker components.. but the @Input(), @Output(), ngOnInit and ngOnDestroy have to be in the icon-picker component either way in order to change icon and color of it's parent.

Also while there is a lot of duplicity at the time, that won't be the case after we add the planned features to it.

**Comment by justmesr** [comment]
peek 2017-07-28 23-27
peek 2017-07-28 23-30

**Comment by kubedo8** [comment]
You commited also my commit. Please remove it.

File: src/main/scripts/src/app/shared/perspectives/collections/post-it/post-it-collections-perspective.component.ts [hunk 7]
```diff
@@ -18,23 +18,119 @@
...
+  public colors: string[] = Const.colors;
+  public newCollections: CollectionModel[] = [];
+  public collections: Collection[];
+  public numbers: any[];
+  public cachedName: string;
+  public selectedIcon: string;
+  public selectedColor: string;
+
   constructor(private collectionService: CollectionService) {
+    this.numbers = Array.apply(null, {length: this.iconsPerPage}).map(Number.call, Number);
```
Comment (kubedo8) [inline src/main/scripts/src/app/shared/perspectives/collections/post-it/post-it-collections-perspective.component.ts]: This is no longer needed.

File: src/main/scripts/src/app/shared/perspectives/collections/post-it/post-it-collections-perspective.component.scss [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+  .remove-icon {
+    cursor: pointer;
+    position: absolute;
+    color: red;
+    right: 10px;
+    top: 10px;
+  }
+
+  .color-icon {
+    z-index: 9;
```
Comment (kubedo8) [inline src/main/scripts/src/app/shared/perspectives/collections/post-it/post-it-collections-perspective.component.scss]: You should remove unneccesary scss classes after removing old color-icon picker

File: src/main/scripts/src/app/shared/perspectives/icon-picker/palette/icon-picker-palette.component.ts [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+ * limitations under the License.
+ * -----------------------------------------------------------------------/
+ */
+
+import {Component, EventEmitter, Input, OnDestroy, OnInit, Output} from '@angular/core';
+
+@Component({
+  selector: 'palette',
+})
+export class PaletteComponent implements OnInit, OnDestroy {
```
Comment (livthomas) [inline src/main/scripts/src/app/shared/perspectives/icon-picker/palette/icon-picker-palette.component.ts]: Is there any intention to use this class alone? If not, it should be made abstract and it does not need `@Component` decorator.

File: src/main/scripts/src/app/shared/perspectives/icon-picker/icon-picker.component.ts [hunk 10]
```diff
@@ -44,132 +43,36 @@ import {LocalService} from '../../../core/local.service';
     ])
   ]
 })
-export class IconPickerComponent implements OnInit, OnDestroy {
+export class IconPickerComponent {
+
+  @Input()
+  public enabled: boolean;
```
Comment (justmesr) [inline src/main/scripts/src/app/shared/perspectives/icon-picker/icon-picker.component.ts]: This way the component can handle animations for itself. (Without it, the instance *ngIf changes to false, the component dissapears without closing animation.

**Pull request merged**